### ISSUE TITLE
New quick fix options to disable the rule for the line, file or the project

### ISSUE THREAD

**Issue by felipecrs** [timonwong/vscode-shellcheck#116]
Example, from [vscode-groovy-lint](https://github.com/nvuillam/vscode-groovy-lint):

![wPRV4yXbXm](https://user-images.githubusercontent.com/29582865/94277469-75386580-fefe-11ea-87f2-706534da308d.gif)

## Disable for the line

Includes a statement in the previous line, such as `# shellcheck disable=SC2035`. [Reference](https://github.com/koalaman/shellcheck/blob/master/shellcheck.1.md#directives).

## Disable for the entire file

Includes a ShellCheck directive statement before the first command in the file.

## Disable for the project

Adds the ShellCheck directive statement to the `.shellcheckrc` file (and create the file if it does not exist).

**Comment by beltaurus** [comment]
We can't just move a directive to the first line of a file, if there is a shebang present (which *has* to be the first line, nay, the first characters of that file!).

In my case, even if I don't have a shebang as the first line, I have a directive to indicate the style of the script:
# shellcheck shell=bash

I believe that a directive like that also has to be on the first line (and a shebang makes for not needing this directive.

Next, there are corporate directives on where to put licensing information, for a lot of the bigger companies. This accounts for the next few lines (typically a comment). These lines can also hold revision control information... (for older revision control systems)...

Let's think about a more intuitive way, and maybe a way that works for more customer scenarios?

**Comment by felipecrs** [comment]
Right, I said exactly what koalaman said in the ShellCheck documentation:

> Includes a ShellCheck directive statement before the first command in the file.

I believe we should simply find what is the first command in the file, and put the directives before. :)

**Comment by beltaurus** [comment]
If this is what you mean:
```if (first_line == (shebang || directive) then
  put_new_directive_on second_line
else
  put_new_directive_on_second_line
end
```
Maybe...

What about:
```
skip_all_comment_lines at the start of the file();
insert_new_directive()
```

The latter seems to be safer and more comprehensive than the first?
And definitely KISS!

**Comment by felipecrs** [comment]
That was not what I meant... when koalaman said "command" I believe he meant anything except "comments". 

```

Regarding:

> ```
> skip_all_comment_lines at the start of the file();
> insert_new_directive()
> ```

I prefer to take an approach closer to the statement:

``` javascript
find_first_command_in_the_file(); // this implies skipping all the comments
insert_new_directive_one_line_above_it() 
```

**Comment by felipecrs** [comment]
E.g.:

````bash
#!/bin/bash

# Copyright Felipe
# Dummy line

# Another comment

# Usage: hello-world

echo "hello-world"
```

would become:

````bash
#!/bin/bash

# Copyright Felipe
# Dummy line

# Another comment

# Usage: hello-world

# shellcheck disable=SC2035

echo "hello-world"
```

**Comment by dcd-arnold** [comment]
I would like to have at least the line-based action. It is really a pain to add the comment.

**Comment by felipecrs** [comment]
Now you have snippets BTW:

https://github.com/vscode-shellcheck/vscode-shellcheck/pull/1190

**Comment by DanKaplanSES** [comment]
> I would like to have at least the line-based action. It is really a pain to add the comment.

I agree. 

@felipecrs Other downsides to extracting `Disable for the line` as a separate issue?

**Comment by felipecrs** [comment]
No downsides, but it's probably not necessary. PRs are welcome regardless of being only for the line or not.

**Comment by DanKaplanSES** [comment]
Ah, my mistake. I assumed you wanted a PR to implement all three.
